src/lib/list/list.ts

```diff
@@ -165,12 +196,24 @@ export class MatListItem extends _MatListItemMixinBase implements AfterContentIn
     if (element.nodeName.toLowerCase() === 'button' && !element.hasAttribute('type')) {
       element.setAttribute('type', 'button');
     }
+
+    // @breaking-change 8.0.0 Remove null check for _changeDetectorRef.
+    if (this._list && _changeDetectorRef) {
+      this._list._stateChanges.pipe(takeUntil(this._destroyed)).subscribe(() => {
```

Comment:
Worth adding a comment in what situations this can happen? e.g. if the ripple input from the actual list changes.

Reply:
My thoughts were that it's not strictly necessary since the unit tests capture the behavior already.

Reply:
True, it's not necessary, but I kind of doubt that someone can reason `stateChanges` by looking at these tests. I don't feel too strong about it. We can leave it out if you feel it's not necessary.